mimic/rest/loadbalancer_api.py

```diff
@@ -167,6 +167,48 @@ def set_attributes(self, request, tenant_id, clb_id):
             request.setResponseCode(204)
             return b''
 
+    @app.route(
+        '/v2/<string:tenant_id>/loadbalancer/<int:clb_id>/error_reason',
+        methods=['POST']
+    )
+    def set_error_reason(self, request, tenant_id, clb_id):
+        """
+        Sets a configurable response code and message that will be returned
+        when the load balancer is in an ERROR state.
```

Comment:
I'm a little concerned about providing super flexible APIs like this.

One of Mimic's purposes is to provide a flexible test backend.  However, Mimic should also serve as a repository of useful error responses; things that we know can happen for documented reasons, and keep those responses updated as backend services change.

For example, I think it was a mistake to make the Nova mock's "failure" behavior so flexible (allowing an arbitrary status code and message); Nova cannot in reality return anything from anywhere, it has a fixed (albeit large) set of failure modes.  One mode I'd really like to see from Mimic in the future is a "fuzz mode" where it returns a series of realistic errors, and we won't be able to do that until we have a good selection of un-parameterized failures.

There are tickets for this - #270 for doing it for CLBs, #269 for doing it for Nova.

I wouldn't block this from going in on this basis (Nova already has one too-flexible failure API, it would be better to have *something* for CLBs).  But, as I understand it, the way that test case development is going to go is that you will identify some failure cases that need testing, and then go into Autoscale's test suite and add each expected failure from Nova or from CLBs into calls to mimic from Autoscale.

It would be good if we could, instead of doing that, each time an Autoscale test case needs a new error from CLBs or from Nova, instead of hitting a generalized "failure" endpoint for each new thing, make a Mimic PR for each new thing, so that they can be classified, named, and documented as a shared resource.  This way we can also have a separate artifact to show to the various teams (CLBs, Nova, etc) to say "this is the failure mode we think you've described to us, did we get it right", and something that others can check against the documentation.  If the parameters are squirreled away in an Otter test case it seems like it would be harder to get multiple teams to look at and update them.